jobbergate-agent/jobbergate_agent/jobbergate/update.py

```diff
@@ -70,6 +82,132 @@ async def update_job_data(
         response.raise_for_status()
 
 
+async def fetch_influx_data(
+    time: int, host: str, step: int, task: int, job: int, measurement: INFLUXDB_MEASUREMENT
+) -> list[InfluxDBMeasure]:
+    """
+    Fetch data from InfluxDB for a given host, step and task.
+    """
+    query = f"""
+    SELECT * FROM {measurement} WHERE time > $time AND host = $host AND step = $step AND task = $task AND job = $job
+    """
+    with JobbergateApiError.handle_errors("Failed to fetch data from InfluxDB", do_except=log_error):
+        assert INFLUXDB_CLIENT is not None
+        params = dict(time=time, host=host, step=str(step), task=str(task), job=str(job))
+        logger.debug(f"Querying InfluxDB with: {query=}, {params=}")
+        result = INFLUXDB_CLIENT.query(query, bind_params=params, epoch="us")
+        logger.debug("Successfully fetched data from InfluxDB")
+        return [
```

Comment:
This could be changed to an iterator. I hope that we can reduce memory footprint by avoiding this big list.

Reply:
in my latest commit ([344042a](https://github.com/omnivector-solutions/jobbergate/pull/663/commits/344042a07950c57cdf9335bc08e454f166bcab99)) I've made changes to use `numba` and `numpy` for these operations. Have a check please.